plugins/rating/services/RatingService.php

```diff
@@ -0,0 +1,156 @@
+<?php
+/**
+ * Allows user to manipulate their entry rating
+ *
+ * @service rating
+ * @package plugins.rating
+ * @subpackage api.services
+ */
+
+class RatingService extends KalturaBaseService
+{
+	public function initService($serviceId, $serviceName, $actionName)
+	{
+		parent::initService($serviceId, $serviceName, $actionName);
+		
+		if(!RatingPlugin::isAllowedPartner($this->getPartnerId()))
+		{
+			throw new KalturaAPIException(KalturaErrors::FEATURE_FORBIDDEN, RatingPlugin::PLUGIN_NAME);
+		}
+		
+		if ((!kCurrentContext::$ks_uid || kCurrentContext::$ks_uid == "") && $actionName != "getRatingCounts")
+		{
+			throw new KalturaAPIException(KalturaErrors::INVALID_USER_ID);
+		}
+	}
+	
+	/**
+	 * @action rate
+	 * Action for current kuser ro rate a specific entry
+	 * @param string $entryId
+	 * @param int $rank
+	 * @throws KalturaErrors::ENTRY_ID_NOT_FOUND
+	 * @throws KalturaErrors::INVALID_RANK_VALUE
+	 * @return int
+	 */
+	public function rateAction ($entryId, $rank)
+	{
+		if (!$entryId)
+		{
+			throw new KalturaAPIException(KalturaErrors::MISSING_MANDATORY_PARAMETER, "entryId");
+		}
+		
+		$dbEntry = entryPeer::retrieveByPK($entryId);
+		if (!$dbEntry)
+		{
+			throw new KalturaAPIException(KalturaErrors::ENTRY_ID_NOT_FOUND, $entryId);
+		}
+		
+		//Check if a kvote for current entryId and kuser already exists.
+		$existingKVote = kvotePeer::doSelectByEntryIdAndPuserId($entryId, $this->getPartnerId(), kCurrentContext::$ks_uid);
+		if ($existingKVote)
+		{
+			$existingKVote->setRank($rank);
+			$existingKVote->save();
+		}
+		else
+		{
+			kvotePeer::createKvote($entryId, $this->getPartnerId(), kCurrentContext::$ks_uid, $rank, KVoteType::RANK);
+		}
+		
+		return $rank;
+	}
+	
+	/**
+	 * @action removeRating
+	 * Action to remove current kuser's rating for a specific entry
+	 * @param string $entryId
+	 * @throws KalturaErrors::ENTRY_ID_NOT_FOUND
+	 * @throws KalturaRatingErrors::USER_RATING_FOR_ENTRY_NOT_FOUND
+	 * @return bool
+	 */
+	public function removeRatingAction ($entryId)
+	{
+		if (!$entryId)
+		{
+			throw new KalturaAPIException(KalturaErrors::MISSING_MANDATORY_PARAMETER, "entryId");
+		}
+		
+		$dbEntry = entryPeer::retrieveByPK($entryId);
+		if (!$dbEntry)
+		{
+			throw new KalturaAPIException(KalturaErrors::ENTRY_ID_NOT_FOUND, $entryId);
+		}
+		
+		$existingKVote = kvotePeer::doSelectByEntryIdAndPuserId($entryId, $this->getPartnerId(), kCurrentContext::$ks_uid);
+		if (!$existingKVote)
+		{
+			throw new KalturaAPIException(KalturaRatingErrors::USER_RATING_FOR_ENTRY_NOT_FOUND);
+		}
+		
+		
+		if (kvotePeer::disableExistingKVote($entryId, $this->getPartnerId(), kCurrentContext::$ks_uid))
+			return true;
+		
+		return false;
```

Comment:
Would not it be more efficient to return the new rand as in ratingAction

Reply:
@MosheMaorKaltura  but we're **removing** the rank.